include/onnxruntime/core/session/onnxruntime_cxx_api.h

```diff
@@ -4,13 +4,23 @@
 // Summary: The Ort C++ API is a header only wrapper around the Ort C API.
 //
 // The C++ API simplifies usage by returning values directly instead of error codes, throwing exceptions on errors
-// and automatically releasing resources in the destructors.
+// and automatically releasing resources in the destructors. The primary purpose of C++ API is exception safety so
+// all the resources follow RAII and do not leak memory.
 //
 // Each of the C++ wrapper classes holds only a pointer to the C internal object. Treat them like smart pointers.
-// To create an empty object, pass 'nullptr' to the constructor (for example, Env e{nullptr};).
+// To create an empty object, pass 'nullptr' to the constructor (for example, Env e{nullptr};). However, you can't use them
+// until you assign an instance that actually holds an underlying object.
 //
-// Only move assignment between objects is allowed, there are no copy constructors. Some objects have explicit 'Clone'
-// methods for this purpose.
+// For Ort objects only move assignment between objects is allowed, there are no copy constructors.
+// Some objects have explicit 'Clone' methods for this purpose.
+//
+// ConstXXXX types are copyable since they do not own the underlying C object, so you can pass them to functions as arguments
+// by value or by reference. ConstXXXX types are restricted to const only interfaces.
+//
+// UnownedXXXX are similar to ConstXXXX but also allow non-const interfaces.
+//
+// The life span of Const/UnownedXXX types must eclipse the lifespan of the owning C++ wrapper object. They exists so you do not
```

Comment:
should it be that the lifetime of the corresponding owning object must eclipse the lifetimes of the Const/Unowned types?
would it be more accurate to say that one should not use (aside from destroying) the Const/Unowned type after the corresponding owner has gone out of scope?